gpjax/dataset.py

```diff
@@ -31,8 +30,8 @@ class Dataset(Pytree):
         y (Optional[Float[Array, "N Q"]]): Output data.
     """
 
-    X: Optional[Float[Array, "N D"]] = None
-    y: Optional[Float[Array, "N Q"]] = None
+    X: Optional[Union[Num[Array, "N D"], Num[Array, "..."]]] = None
+    y: Optional[Union[Num[Array, "N Q"], Num[Array, "..."]]] = None
```

Comment:
I don't remember why I put it in but the dataset fields should be consistent and always be "N D" / "N Q" even if D or Q are 1?

Reply:
on second thought it might have been so that it actually gets to the `_check_shape` in the post_init (otherwise beartype errors at the constructor already). maybe let's just leave it that way:
```suggestion
    X: Optional[Num[Array, "N D"]] = None
    y: Optional[Num[Array, "N Q"]] = None
```

Reply:
does it make sense that they're optional, actually?